test/client/spec/victory-util/collection.spec.js

```diff
@@ -174,4 +177,104 @@ describe("collections", () => {
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it("returns early when nested collections are strictly equal", () => {
+      const a = { test: { nested: "a" } };
+      const b = a;
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns early when nested collections are not the same length", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "b" } };
+      const b = { test: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early if values are not the same type", () => {
+      const a = { test: { nested: "a" } };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early when nested elements are empty", () => {
+      const a = { a: [] };
+      const b = { a: [] };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns false for mixed objects and arrays", () => {
+      const a = { a: [] };
+      const b = { a: {} };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns true if values are functions", () => {
+      const a = { test: () => "a" };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+    it("recursively checks equality for nested collections", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "a" } };
+      const b = { test: { nested: "a" }, test2: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledThrice;
+    });
+
+    it("compares objects regardless of key order", () => {
+      const a = { test2: { nested: "a" }, test: { nested: "b" } };
+      const b = { test: { nested: "b" }, test2: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+    it("does not recursively check date objects", () => {
+      const a = { test: new Date(2010, 2, 1), test2: new Date(2010, 1, 1) };
+      const b = { test: new Date(2010, 2, 1), test2: new Date(2010, 1, 1) };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns equal for equivalent date objects", () => {
+      const a = { test: new Date(2010, 1, 1) };
+      const b = { test: new Date("Mon Feb 01 2010") };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+    const pairs = [
```

Comment:
We need to either (1) attribute the lodash license and make sure it's compatible with Victory's for this literal copy and paste, OR (2) use it for inspiration / cases but not copy and paste it. Either way is fine.